config/src/main/java/org/springframework/security/config/annotation/web/configuration/WebMvcSecurityConfiguration.java

```diff
@@ -72,6 +78,9 @@ RequestDataValueProcessor requestDataValueProcessor() {
 	@Override
 	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
 		this.beanResolver = new BeanFactoryResolver(applicationContext.getAutowireCapableBeanFactory());
+		if (applicationContext.getBeanNamesForType(SecurityContextHolderStrategy.class).length > 0) {
+			this.securityContextHolderStrategy = applicationContext.getBean(SecurityContextHolderStrategy.class);
```

Comment:
Will this fail with multiple beans of SecurityContextHolderStrategy?

Reply:
Yes, this will fail and that was my intent. However, Rob and I discussed and the prevailing pattern in Spring Security is to return null if there are multiple of the same bean, allowing the application to fall back to the defaults. This may be worth discussing in the future.